src/XMakeBuildEngine/Construction/Solution/SolutionProjectGenerator.cs

```diff
@@ -1942,6 +1917,13 @@ private void AddTraversalTargetForProject(ProjectInstance traversalProject, Proj
                 actualTargetName += ":" + targetToBuild;
             }
 
+            // Don't add the target again.  The user might have specified /t:Project:target which was already added but only this method knows about Project:Target so
+            // after coming up with that target name, it can check if it has already been added.
+            if (traversalProject.Targets.ContainsKey(actualTargetName))
```

Comment:
If the project name contains a `.` or if you have two projects with the same name (like foo.csproj and foo.vbproj) then only this method knows what the final target names will be.  

I tried to explain that in the comment.  Should I reword it so it makes more sense?